core/embed/firmware/bootloader_hashes.py

```diff
@@ -29,7 +29,7 @@ def aligned_digest(data: bytes, padding: bytes) -> bytes:
     (unwritten NOR-flash byte) or 0x00 (explicitly cleared byte).
     """
     if len(data) > ALIGNED_SIZE:
-        raise ValueError(fn, "too big")
+        raise ValueError(len(data), "too big")
```

Comment:
Let's also keep the filename, so it is obvious from the error message which file failed the assertion.
```suggestion
        raise ValueError(fn, len(data), "too big")
```

Reply:
I removed the `fn`, because it is undefined here (flagged by `ruff`), we would need to pass it in as an argument

Reply:
Ah, right, nevermind. So this never worked as intended. Let's keep the change as it is.

Reply:
Theoretically the check can be moved to the higher function, there it has access to the filename:
```sh
index 4e90e0f4f0..91cc186d09 100755
--- a/core/embed/firmware/bootloader_hashes.py
+++ b/core/embed/firmware/bootloader_hashes.py
@@ -28,9 +28,6 @@ def aligned_digest(data: bytes, padding: bytes) -> bytes:
     Firmware needs to check the bootloader against a digest padded either by 0xff
     (unwritten NOR-flash byte) or 0x00 (explicitly cleared byte).
     """
-    if len(data) > ALIGNED_SIZE:
-        raise ValueError(len(data), "too big")
-
     assert len(padding) == 1
     digest_data = data + padding * (ALIGNED_SIZE - len(data))
     assert len(digest_data) == ALIGNED_SIZE
@@ -54,6 +51,9 @@ def bootloader_str(file: Path) -> str:
     """
     data = file.read_bytes()
 
+    if len(data) > ALIGNED_SIZE:
+        raise ValueError(file, len(data), "data too big")
+
     suffix = file.stem[len("bootloader_") :].upper()
     bytes_00 = to_uint_array(aligned_digest(data, b"\x00"))
     bytes_ff = to_uint_array(aligned_digest(data, b"\xff"))
```

Reply:
more like, leave this alone and wrap line 87 in a try-except adding the file info

Reply:
Done like this in the rebased force-push